pallets/route-executor/src/lib.rs

```diff
@@ -851,3 +853,31 @@ impl<T: Config> RouteProvider<T::AssetId> for Pallet<T> {
 		}
 	}
 }
+impl<T: Config> RouteSpotPriceProvider<T::AssetId> for Pallet<T> {
+	fn spot_price(route: &[Trade<T::AssetId>]) -> Option<FixedU128> {
+		let mut prices: Vec<FixedU128> = Vec::with_capacity(route.len());
+		for trade in route {
+			let spot_price_result = T::AMM::calculate_spot_price_with_fee(trade.pool, trade.asset_in, trade.asset_out);
+
+			match spot_price_result {
+				Ok(spot_price) => prices.push(spot_price),
+				Err(_) => return None,
+			}
+		}
+		if prices.is_empty() {
+			return None;
+		}
+
+		let nominator = prices.iter().try_fold(U512::from(1u128), |acc, price| {
```

Comment:
you can have a bad time when the route is more than 4 hops. you should probably aggregate if after every 4 hops